crates/core/src/ring/connection_manager.rs

```diff
@@ -4,11 +4,27 @@ use crate::topology::{Limits, TopologyManager};
 
 use super::*;
 
+/// State of a peer connection within the ring topology
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ConnectionState {
```

Comment:
Let's do this, but we should research why we need this in the first place. IMO we should be leaking transient connections into the connection_manager, if we can help it. I think is gonna create other problems.

Transient connections are transient while they establish connections to others, and that's it, we shouldn't consider them for other logical operations.